components/epaxos/src/replica/replica.rs

```diff
@@ -78,13 +81,20 @@ impl From<(ReplicaId, String, bool)> for ReplicaPeer {
     }
 }
 
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub enum ExecuteResult {
+    Success,
+    SuccessWithVal(Option<Vec<u8>>),
+}
+
 /// structure to represent a replica
 pub struct Replica {
     pub replica_id: ReplicaId,
     pub group_replica_ids: Vec<ReplicaId>,
     pub peers: Vec<ReplicaPeer>,
     pub storage: Storage,
     pub committed_timeout: i32,
+    pub reply_txes: Mutex<HashMap<InstanceId, Sender<Vec<ExecuteResult>>>>,
```

Comment:
换这个名字咋样? `waiting_replies`, tx的意思应该能从类型看出来了, 名字感觉主要用来表示业务上的含义好点.
```suggestion
    pub waiting_replies: Mutex<HashMap<InstanceId, Sender<Vec<ExecuteResult>>>>,
```